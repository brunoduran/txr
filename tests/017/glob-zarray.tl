(load "../common")

(defstruct glob-t nil
  pathc pathv (reserve 0))

(cond
  ((eq (os-symbol) :macos)
   (deffi-type glob-t (struct glob-t
                        (pathc size-t)
                        (nil int)
                        (nil size-t)
                        (nil int)
			(pathv (ptr-out-d (zarray str-d)))
                        (nil (array 4 cptr)))))
  (t (deffi-type glob-t (struct glob-t
                          (pathc size-t)
                          (pathv (ptr-out-d (zarray str-d)))
                          (reserve size-t)))))

(with-dyn-lib (libc)
  (deffi globb "glob" int (str int closure (ptr-out glob-t)))
  (deffi-cb glob-cb int (str int) -1))

(let* ((g (new glob-t)))
  (prinl (globb "tests/001/*.txr" 0 (glob-cb (lambda (path err))) g))
  (prinl g))
