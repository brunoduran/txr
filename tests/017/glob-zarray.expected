0
#S(glob-t pathc 4 pathv #("tests/001/query-1.txr" "tests/001/query-2.txr" "tests/001/query-3.txr"
                          "tests/001/query-4.txr") reserve 0)
